test/multipoint_routes.cc

```diff
@@ -348,6 +348,36 @@ TEST(MultiPointRoutesBreakThrough, test_mid_break_through_arrive_by) {
   test_mid_break_through(R"(,"date_time":{"type":2,"value":"2016-07-03T08:06"}})");
 }
 
+TEST(MultiPointRoutesBreakThrough, test_mid_break_through_elapsed) {
+  route_tester tester;
+  std::string request =
+      R"({"locations":[
+            {"lat": 52.1191920, "lon":5.1026068, "type": "break_through", "node_snap_tolerance": 0},
+            {"lat": 52.1189946, "lon": 5.1029042, "type": "break_through", "node_snap_tolerance": 0},
+            {"lat": 52.1192212, "lon": 5.1024556, "type": "break_through", "node_snap_tolerance": 0}], "costing":"auto"})";
```

Comment:
Any chance you could make these into Gurka tests instead of using Utrecht?  It's a bit hard to visualize exactly what kind of geometry this test is routing over.

Reply:
honestly i think we should rewrite most, if not all, of the utrecht based tests in gurka but id love to tackle that separately as its a huge effort.

specific to this one though is that the bug is triggered by the router expanding the graph from both ends in just the right order that the forward direction expansion has > 1 edges (3 in this case) in it when it reaches the meeting point and the reverse direction has exactly 1 edge in it. i feel like i could spend a long time trying to concoct a graph that will illicit that behavior from the algorithm when i have already a test case here.

i totally understand its hard to tell what is going on without a picture though so id be happy to throw it up on geojson.io and draw the forward edges and the reverse edge to illustrate what is happening. would that work for you?

Reply:
Right - your explanation here is what I think is not coming through in the test, and will get lost once this gets merged.  Can you add this explanation to the test case?

Reply:
of course